### ISSUE TITLE
Error 'field type conflict' while starting InfluxDB

### ISSUE THREAD

**Issue by jbourke** [influxdata/influxdb#8689]
### Bug report

__System info:__ 
Centos7
InfluxDB 1.2.0 wrote the shards
InfluxDB 1.3.2 trying to open the shards

We had been running Influx 1.2.0 and started experiencing some crashes around an issue fixed in 1.2.1 which related to subrequests causing recursion.  Out of this came the need to upgrade, and we moved to 1.3.2.  During our crashes, we had been forced to delete the most recent shard due to corruption to get the system started again.

Since then, we have noticed that newly shards created which had been running fine after their creation could no longer be loaded if Influx was restarted.  We had been trying to restore data from disk snapshot backups, and realized that these particular shards would not load at during Influx startup.

The attached log has the entire load up sequence of a restored volume featuring these lines:
```
Failed to open shard: 6663: [shard 6663] field type conflict service=store
Failed to open shard: 6861: [shard 6861] field type conflict service=store
```
The issue is that nothing is being written yet; this occurs during a shard load before the server even opens endpoint ports.  We have been through all the tools and docs we can find looking to explain this issue, and while this error is mentioned frequently, it is dismissed as a write error on the users part which is not our case.  This occurs during load, not during a write.

Is there any background, explanation, or tool to better investigate this issue?  To summarize:

- New shard is created and appears to run fine while Influx is up
- Shard takes data for several days with no issues experienced
- Gracefully stop Influx
- Start Influx, shard(s) can not be loaded due to type conflict

__Additional info:__ 
[influxdb-shard-crash.txt](https://github.com/influxdata/influxdb/files/1217994/influxdb-shard-crash.txt)

**Comment by jwilder** [comment]
That error is occurring because you have somehow managed to get a shard that is storing two different field types for the same field (e.g. `cpu value=0` and `cpu value=0i` `float64` vs `int64`).  Field types can be different across shards, but must be consistent within a shard.  You shouldn't be able to get into this state, but somehow you have which would be a bug.

There are two things you said you did that might have allowed this to occur inadvertently.

1. Deleting data -  There was a bug when deleting measurements/series where the index could become inconsistent with the shard on disk (#8576).  If the shard was in this state, it may allow a write for a field with a different type.  You'd see the error at restart.  If you ran drop shard to remove the shard, this probably wouldn't be possible though.  This could definitely occur in any release prior to 1.3.2.
2. Restoring data into shard - If you restored data into an existing shard with data where the field type had been different in the old version, it might be possible to get into this state as well.  If you manually copied TSM files into a shard, it definitely could cause this issue as well.

As a workaround, you could move shard 6663 and 6861 dirs out of the data and wal dirs so your server can start up.  You can then run `influx_inspect export` to export the data from those shards into a format that you can use with `influx -import` to replay the data.  You can add `-h` to either of those commands see the available options as you will likely need to specify `-datadir` and `-waldir` and possibly the `-database` and `-retention`.

**Comment by jwilder** [comment]
#8701 fixes an issue w/ `drop measurement` that could cause this error as well.  I was able to repro it locally.

**Comment by jwilder** [comment]
This should be fixed via #8701 and #8576.
